### ISSUE TITLE
Reporting for Rack applications does not work

### ISSUE THREAD

**Issue by mkarganov** [rollbar/rollbar-gem#1031]
Hi! We are using Hanami and integrating rollbar there. I noticed that due to [this code](https://github.com/rollbar/rollbar-gem/blob/63c68eed6a8066cdd8e09ab5429728d187482b12/lib/rollbar/middleware/rack.rb#L48) reports are never executed for Hanami (most likely for other Rack applications too).

I believe simple change to:

```
      def framework_error(env)
        env['rack.exception']
      end
```

should do the job. Or at least defining similar middleware as for [Sinatra](https://github.com/rollbar/rollbar-gem/blob/master/lib/rollbar/middleware/sinatra.rb#L5).

**Comment by waltjones** [comment]
I'm not sure the current code isn't correct.

Uncaught exceptions will be reported here:
https://github.com/rollbar/rollbar-gem/blob/63c68eed6a8066cdd8e09ab5429728d187482b12/lib/rollbar/middleware/rack.rb#L24-L26

`env['sinatra.error']` is a feature of Sinatra, so the logic in the Sinatra middleware is supporting that. I'm not sure a similar thing has been standardized for bare Rack apps.

If the exception isn't reaching the rescue clause in the Rollbar middleware, it might be because Hanami catches it first. It looks like Hanami puts its global error handler in the controller:
https://guides.hanamirb.org/actions/exception-handling/

Implementing a custom error handler there that includes a call to `Rollbar.error(e)` is probably the best solution.

**Comment by mkarganov** [comment]
Thank you for fast response. I think it is worth checking.

You are right that Hanami catches it here: https://github.com/hanami/controller/blob/master/lib/hanami/action/rack/errors.rb#L23

Isn't having env['rack.exception'] for Rack apps in case of an error is some kind of standard?
(just for reference there is similar place from honeybadger: https://github.com/honeybadger-io/honeybadger-ruby/blob/master/lib/honeybadger/rack/error_notifier.rb#L87).

I also believe that more proper way is to report exceptions via Rack Middleware rather then writing custom handlers as they are a bit more specific and exceptions should be reported on a more generic level.

Anyway it would be nice to have the support of it in rollbar as other gems have 👍🏻

**Comment by waltjones** [comment]
You're right, I suppose some people are using `env['rack.exception']`. This seems reasonable to add.

**Comment by mkarganov** [comment]
Thanks! I was looking into some rack standards, but could not find any. 

Do you see it as a change to `framework_exception` method?

If it takes some time, probably we just monkey patch it for now.

**Comment by waltjones** [comment]
I also haven't seen this documented anywhere, but since it is widely used, I think your original suggestion to add it to `framework_error` should be fine.

The next release could be a while, so yes you might want to patch it for now.
